store/reducers/quizes.js

```diff
@@ -0,0 +1,75 @@
+import * as firebase from 'firebase';
+import 'firebase/firestore';
+
+export const FETCH_QUIZES = 'FETCH_QUIZES';
+export const FETCH_STATUS = 'FETCH_STATUS';
+export const SAVE_QUIZ = 'SAVE_QUIZ';
+
+const INITIAL_STATE = {
+  quizes: [],
+  isFetching: false,
+};
+
+export default function quizesReducer(state = INITIAL_STATE, action) {
+  switch (action.type) {
+    case FETCH_QUIZES:
+      return {
+        ...state,
+        quizes: action.quizes,
+      };
+    case FETCH_STATUS:
+      return {
+        ...state,
+        isFetching: action.isFetching,
+      };
+    case SAVE_QUIZ:
+      return {
+        ...state,
+        quizes: [action.quiz, ...state.quizes],
+      };
+    default:
+      return { ...state };
+  }
+}
+
+export const fetchQuizes = () => {
+  return async (dispatch, getState) => {
+    const { user } = getState();
+    dispatch({ type: FETCH_STATUS, isFetching: true });
+    await firebase
+      .firestore()
+      .collection('quizResults')
+      .where('uid', '==', user.user.uid)
```

Comment:
what happens if a user does not have quizResults collection yet? Would this fail out on the where clause?

Reply:
Just logged in on as a test user and it's all good. Think it just returns an empty array - doesn't error out.